ndcube/tests/test_ndcubeio.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+
+# Tests to perform for to_hdulist
+# Test HDU0
+    # Test the meta
+    # Test the wcs axes
+    # Test the missing_axes
+# Test HDU1
+    # Test the units
+# Test HDU2
+    # Test the uncertainty
+# Test HDU3
+    # Test the mask
+# Test HDU4
+    # Test the extra_coords
+import pytest
+import numpy as np
+
+import astropy.units as u
+from astropy.io import fits
+# Import the NDCube objects from test_ndcube rather than defining them out here
+# from ndcube.tests import test_ndcube
+from ndcube.mixins.ndcubeio import NDCubeIOMixin
+from ndcube import NDCube
+from ndcube.utils.wcs import WCS
+from ndcube.tests.helpers import comparerecords
+
+
+# The NDCube objects
+data = np.array([[[1, 2, 3, 4], [2, 4, 5, 3], [0, -1, 2, 3]],
+                 [[2, 4, 5, 1], [10, 5, 2, 2], [10, 3, 3, 0]]])
+
+hm = {'CTYPE1': 'WAVE    ', 'CUNIT1': 'Angstrom', 'CDELT1': 0.2, 'CRPIX1': 0, 'CRVAL1': 10,
+      'NAXIS1': 4,
+      'CTYPE2': 'HPLT-TAN', 'CUNIT2': 'deg', 'CDELT2': 0.5, 'CRPIX2': 2, 'CRVAL2': 0.5,
+      'NAXIS2': 3,
+      'CTYPE3': 'HPLN-TAN', 'CUNIT3': 'deg', 'CDELT3': 0.4, 'CRPIX3': 2, 'CRVAL3': 1, 'NAXIS3': 2}
+wm = WCS(header=hm, naxis=3)
+uncertaintym = data
+mask_cubem = data > 0
+meta = {"Description": "This is example NDCube metadata."}
+cubem = NDCube(
+    data,
+    wm,
+    mask=mask_cubem,
+    meta=meta,
+    uncertainty=uncertaintym,
+    extra_coords=[('time', 0, u.Quantity(range(data.shape[0]), unit=u.pix)),
+                  ('hello', 1, u.Quantity(range(data.shape[1]), unit=u.pix)),
+                  ('bye', 2, u.Quantity(range(data.shape[2]), unit=u.pix))])
+
+
+# HDUList object
+cubem_hdulist = NDCubeIOMixin.to_hdulist(cubem)
+
+# Testing the HDU0
+@pytest.mark.parametrize(
+    "header_obj, header_values1, header_values2, header_values3, header_values4, header_values5, missing_axes",[
+        ((cubem_hdulist[0].header),(3, 0.0, 2.0, 2.0),(2E-11, 0.5, 0.4), ('m','deg','deg'),('WAVE','HPLT-TAN','HPLN-TAN'),(1E-09, 0.5, 1.0),(0, 0, 0))
+    ]
+)
+def test_meta_from_hdulist(header_obj, header_values1, header_values2, header_values3, header_values4, header_values5, missing_axes):
+    wcs_axes, crpix1, crpix2, crpix3 = header_values1
```

Comment:
Here I haven't found a way to test meta values, so maybe we need to decide on the way we store meta values? So that we have only **one** way to access them.